runtime/tests/interpreter/entitlements_test.go

```diff
@@ -2972,3 +2972,111 @@ func TestInterpretBuiltinEntitlements(t *testing.T) {
 	_, err := inter.Invoke("main")
 	assert.NoError(t, err)
 }
+
+func TestInterpretIdentityMapping(t *testing.T) {
+
+	t.Parallel()
+
+	t.Run("owned value", func(t *testing.T) {
+		t.Parallel()
+
+		inter := parseCheckAndInterpret(t, `
+            struct S {
+                access(Identity) fun foo(): auth(Identity) &AnyStruct {
+                    let a: AnyStruct = "hello"
+                    return &a as auth(Identity) &AnyStruct
+                }
+            }
+
+            fun main() {
+                let s = S()
+
+                // OK: Must return an unauthorized ref
+                let resultRef1: &AnyStruct = s.foo()
```

Comment:
It doesn't make a lot of sense to me that accessing `foo` on an owned value would give you fewer permissions than accessing it on an entitled reference.

Reply:
but unfortunately, the possible entitlement set is infinite, and currently we don't have a way to represent such an entitlement.

Reply:
Discussed this with Supun, and agree that this is the expected behaviour given the current feature definition.

Maybe in the future we could introduce a notion of "all entitlements", basically something authorized to have the same access as owner-access has. Not sure, but this is maybe what Dete mentioned with "owner" access before.

Reply:
Created issue https://github.com/onflow/cadence/issues/2635 to track this.